CMakeLists.txt

```
@@ -180,3 +180,36 @@ if(USE_UPDATER)
   endif()
 
 endif()
+
+# Get the latest commit hash
+execute_process(
+        COMMAND git rev-parse --short HEAD
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE GIT_COMMIT_HASH
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(GIT_COMMIT_HASH)
+  STRING(TOUPPER ${GIT_COMMIT_HASH} GIT_COMMIT_HASH)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "-DGIT_COMMIT_HASH=\"${GIT_COMMIT_HASH}\"")
+endif()
+
+# Get the current working branch
+execute_process(
+        COMMAND git describe --all --dirty --broken
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE GIT_BRANCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(GIT_BRANCH)
+  STRING(REGEX REPLACE "heads/" "" GIT_BRANCH ${GIT_BRANCH})
+
+  if(GIT_BRANCH MATCHES ".*-dirty$")
+    STRING(REGEX REPLACE "-dirty$" " (DIRTY)" GIT_BRANCH ${GIT_BRANCH})
```

`dirty` to the common player means something else, I suggest that we translate to something more meaningful.

```suggestion
    STRING(REGEX REPLACE "-dirty$" " (modified)" GIT_BRANCH ${GIT_BRANCH})
```